packages/blade/src/components/Input/SelectInput/SelectInput.tsx

```diff
@@ -64,12 +82,88 @@ const _SelectInput = (
     },
   });
 
-  const { icon, onChange, placeholder = 'Select Option', onBlur, ...baseInputProps } = props;
+  const {
+    icon,
+    onChange,
+    defaultValue,
+    placeholder = 'Select Option',
+    onBlur,
+    ...baseInputProps
+  } = props;
+
+  const getValuesArrayFromIndices = (): string[] => {
+    let indices: number[] = [];
+    if (isControlled) {
+      indices = controlledValueIndices;
+    } else {
+      indices = selectedIndices;
+    }
+
+    return indices.map((selectionIndex) => options[selectionIndex].value);
+  };
+
+  const isFirstRenderRef = React.useRef(true);
+
+  const selectValues = (valuesToSelect: string | string[]): void => {
+    if (options.length > 0) {
+      if (typeof valuesToSelect === 'string') {
+        // single select control
+        const selectedItemIndex = options.findIndex((option) => option.value === valuesToSelect);
+        if (selectedItemIndex >= 0) {
+          setSelectedIndices([selectedItemIndex]);
+        }
+      } else {
+        // multiselect control
+
+        // Handles repeated values in user state
+        const uniqueValues = Array.from(new Set(valuesToSelect));
+        // Handle selectionType single with multiselect values
+        const userValues = selectionType === 'single' ? [valuesToSelect?.[0]] : uniqueValues;
+
+        const selectedItemIndices = userValues
+          .map((optionValue) => options.findIndex((option) => option.value === optionValue))
+          .filter((value) => value >= 0);
+
+        setSelectedIndices(selectedItemIndices);
+      }
+    }
+  };
 
+  // Handles `defaultValue` prop
   React.useEffect(() => {
-    onChange?.({ name: props.name, values: value.split(', ') });
+    if (options.length > 0 && props.defaultValue) {
+      selectValues(props.defaultValue);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options.length]);
+
+  // Handles `value` prop
+  React.useEffect(() => {
+    if (options.length > 0 && props.value) {
+      if (!isControlled) {
+        setIsControlled(true);
+      }
+
+      selectValues(props.value);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.value, options]);
+
+  React.useEffect(() => {
+    // Ignore calling onChange on mount
+    if (isFirstRenderRef.current) {
```

Comment:
why do we need to check this explicitly?

Reply:
`useEffect` by default will run on mount. Which means the `onChange` call inside will run on mount which is not expected. We only want to trigger onChange after something has changed and not on mount. This is the part that fixes #1102

Reply:
but onchange shouldn't trigger on mount regardless right?

Reply:
> but onchange shouldn't trigger on mount regardless right?

Ok so normally in other inputs, you can just call the `onChange` callback when `onChange` on the input is triggered. So when user types into TextInput, the onChange of text input works as expected. 

However in Select + Dropdown's case, we're changing values of Select programatically (based on state, when actionlistitem is clicked). So we cannot directly put `onChange` on `BaseInput` of Select because it will never get called (`onChange` only gets triggered when value is changed on user interaction with that input. Not when we programatically set that value).

So we kind of have to simulate the `onChange` behaviour here. So all this useEffect, and callbackTrigerer, first mount ignore is to simulate that behaviour.